Makefile

```diff
@@ -3,81 +3,102 @@ include $(SELF_DIR)/.ci/common.mk
 test_ci_integration := .ci/test-integration.sh
-convert-test-data := .ci/convert-test-data.sh
-coverfile := cover.out
-coverage_xml := coverage.xml
-junit_xml := junit.xml
-test_log := test.log
-lint_check := .ci/lint.sh
-
-BUILD := $(abspath ./bin)
+convert-test-data   := .ci/convert-test-data.sh
+coverfile           := cover.out
+coverage_xml        := coverage.xml
+junit_xml           := junit.xml
+test_log            := test.log
+lint_check          := .ci/lint.sh
+metalint_check      := .ci/metalint.sh
+metalint_config     := .metalinter.json
+metalint_exclude    := .excludemetalint
+
+BUILD           := $(abspath ./bin)
 LINUX_AMD64_ENV := GOOS=linux GOARCH=amd64 CGO_ENABLED=0
-VENDOR_ENV := GO15VENDOREXPERIMENT=1
+VENDOR_ENV      := GO15VENDOREXPERIMENT=1
 
 SERVICES := \
 	m3nsch_server \
 	m3nsch_client \
 
+.PHONY: setup
 setup:
 	mkdir -p $(BUILD)
 
 define SERVICE_RULES
 
+.PHONY: $(SERVICE)
 $(SERVICE): setup
 	@echo Building $(SERVICE)
 	$(VENDOR_ENV) go build -o $(BUILD)/$(SERVICE) ./$(SERVICE)/.
 
+.PHONY: $(SERVICE)-linux-amd64
 $(SERVICE)-linux-amd64:
 	$(LINUX_AMD64_ENV) make $(SERVICE)
 
 endef
 
+.PHONY: services
 services: $(SERVICES)
+
+.PHONY: services-linux-amd64
 services-linux-amd64:
 	$(LINUX_AMD64_ENV) make services
 
 $(foreach SERVICE,$(SERVICES),$(eval $(SERVICE_RULES)))
 
-all: lint test-ci-unit m3nsch_client m3nsch_server
-	@echo "Making all"
-
+.PHONY: lint
 lint:
 	@which golint > /dev/null || go get -u github.com/golang/lint/golint
 	$(VENDOR_ENV) $(lint_check)
 
+.PHONY: metalint
+metalint: install-metalinter
+	@($(metalint_check) $(metalint_config) $(metalint_exclude) && echo "metalinted successfully!") || (echo "metalinter failed" && exit 1)
+
+.PHONY: test-internal
 test-internal:
 	@which go-junit-report > /dev/null || go get -u github.com/sectioneight/go-junit-report
 	@$(VENDOR_ENV) $(test) $(coverfile) | tee $(test_log)
 
+.PHONY: test-xml
 test-xml: test-internal
 	go-junit-report < $(test_log) > $(junit_xml)
 	gocov convert $(coverfile) | gocov-xml > $(coverage_xml)
 	@$(convert-test-data) $(coverage_xml)
 	@rm $(coverfile) &> /dev/null
 
+.PHONY: test
 test: test-internal
 	gocov convert $(coverfile) | gocov report
 
+.PHONY: testhtml
 testhtml: test-internal
 	gocov convert $(coverfile) | gocov-html > $(html_report) && open $(html_report)
 	@rm -f $(test_log) &> /dev/null
 
+.PHONY: test-ci-unit
 test-ci-unit: test-internal
 	@which goveralls > /dev/null || go get -u -f github.com/mattn/goveralls
 	goveralls -coverprofile=$(coverfile) -service=travis-ci || echo -e "\x1b[31mCoveralls failed\x1b[m"
 
+.PHONY: test-ci-integration
 test-ci-integration:
 	@$(VENDOR_ENV) $(test_ci_integration)
 
+.PHONY: clean
 clean:
 	echo Cleaning build artifacts...
 	go clean
 	rm -rf $(BUILD)
 	@rm -f *.html *.xml *.out *.test
 	echo
 
-.DEFAULT_GOAL := test
-.PHONY: test test-xml test-internal testhtml clean all
+.PHONY: all
+all: lint test-ci-unit m3nsch_client m3nsch_server
```

Comment:
I think you forgot to add `metalint` here

Reply:
Unrelated to your change, could you also re-use `services` rather than spelling out `m3nsch_(server|client)`